portal/src/main/webapp/js/src/pancancer_study_summary/pancancer_study_summary_histogram.js

```diff
@@ -67,7 +67,63 @@ function PancancerStudySummaryHistogram()
     var filterCriteriaChanged = function(model) {
     	return model.hasChanged("cancerType") || model.hasChanged("cancerTypeDetailed") || model.hasChanged("minAlteredSamples") ||	model.hasChanged("minTotalSamples");
     }
-    
+
+    /**
+     * Returns font-size based on the study width
+     * @param studyWidth
+     * @returns {string}
+     */
+    var getFontSize = function(studyWidth){
+        return Math.min((studyWidth * .65), 11) + "px";
+    }
+
+    /**
+     * Determines some dynamic settings which the histogram uses:
+     * - paddingLeft
+     * - height
+     * @param histData
+     */
+    var setupHistogram = function (histData){
+        // determine the maximum label length to be used in the histogram
+        var maxLabelLength = 0;
+        for(var i=0; i<histData.length; i++){
+            var curType = histData[i].typeOfCancer;
+            if(curType.length > maxLabelLength) maxLabelLength = curType.length;
+        }
+        // determine the font-size used
+        var studyWidth = getStudyWidth(histData);
+        var fontSize = getFontSize(studyWidth);
+        // determine the space used by the label and calculate the padding en height
+        var fontSpace = maxLabelLength*fontSize.substr(0, fontSize.length-2)*0.66;
+        calculatePaddingLeft(fontSpace);
+        calculateHeight(fontSpace);
+    }
+
+    /**
+     * Calculate the left padding
+     * @param fontSpace
+     */
+    var calculatePaddingLeft = function(fontSpace){
```

Comment:
Just to make the code concise, maybe you can just use the following statement for the calculatePaddingLeft() function? You already initialized paddingLeft with 80. there is no need to define minPaddingLeft again.
paddingLeft = Math.max(Math.cos(0.33*Math.PI) * fontSpace, paddingLeft);